test/e2e/test_sandbox.py

```
@@ -226,3 +256,66 @@ def test_sandbox_run_stdin(sandbox_ctx, tmp_path, agent):
+def test_sandbox_using_llm_endpoint(caplog, agent):
+    # Configure logging for requests
+    caplog.set_level(logging.CRITICAL, logger="requests")
+    caplog.set_level(logging.CRITICAL, logger="urllib3")
+    test_model = "gemma-4:e2b"
+
+    with RamalamaExecWorkspace() as ctx:
+        # Pull model
+        ctx.check_call(["ramalama", "pull", test_model])
+
+        # Serve an API
+        container_name = f"api{''.join(random.choices(string.ascii_letters + string.digits, k=5))}"
```

Ruff (0.15.18)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)